File: benchmarks/python/calculator.py
def main() -> None:
    print("I can add, subtract, multiply and divide.")
    a_text = input("first number: ")
    b_text = input("second number: ")
    op = input("operation (+ - * /): ")
    try:
        x = float(a_text)
        y = float(b_text)
    except ValueError:
        print("those were not numbers")
        return

    try:
        if op == "+":
            print(x + y)
        elif op == "-":
            print(x - y)
        elif op == "*":
            print(x * y)
        elif op == "/":
            if y == 0:
                raise ZeroDivisionError("Cannot divide by zero.")
            print(x / y)
        else:
            print(f"I don't know the operation {op}")
    except ZeroDivisionError as exc:
        print(exc)


if __name__ == "__main__":
    main()

File: benchmarks/rust/calculator.rs
use std::io::{self, Write};

fn ask(prompt: &str) -> String {
    print!("{}", prompt);
    io::stdout().flush().unwrap();
    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();
    line.trim_end().to_string()
}

fn main() {
    println!("I can add, subtract, multiply and divide.");
    let a_text = ask("first number: ");
    let b_text = ask("second number: ");
    let op = ask("operation (+ - * /): ");
    let first = a_text.parse::<f64>().ok();
    let second = b_text.parse::<f64>().ok();
    if first.is_none() || second.is_none() {
        println!("those were not numbers");
        return;
    }
    let x = first.unwrap();
    let y = second.unwrap();

    match op.as_str() {
        "+" => println!("{}", x + y),
        "-" => println!("{}", x - y),
        "*" => println!("{}", x * y),
        "/" => {
            if y == 0.0 {
                println!("Cannot divide by zero.");
            } else {
                println!("{}", x / y);
            }
        }
        _ => println!("I don't know the operation {}", op),
    }
}
